extensions/AbuseFilter/AbuseFilterVariableHolder.php

```diff
@@ -411,4 +397,38 @@ function compute( $vars ) {
 		return $result instanceof AFPData
 			? $result : AFPData::newFromPHPVar( $result );
 	}
+
+	private function getRecentAuthorsNames( $articleId, $cutOff ): array {
+		$dbr = wfGetDB( DB_SLAVE );
+		$res = $dbr->select( 'revision', 'DISTINCT rev_user, rev_user_text', [
+			'rev_page' => $articleId,
+			'rev_timestamp<' . $dbr->addQuotes( $dbr->timestamp( $cutOff ) ),
+		], __METHOD__, [ 'ORDER BY' => 'rev_timestamp DESC', 'LIMIT' => 10 ] );
+
+		return $this->getUsernamesFromResults( $res );
+	}
+
+	private function getUsernamesFromResults( ResultWrapper $res ): array {
+		$names = User::whoAre( $this->getUsersIds( $res ) );
+		$res->rewind();
+		$users = [];
+		foreach ( $res as $row ) {
+			$users[] = $row->rev_user && isset( $names[$row->rev_user] ) ?
+				$names[$row->rev_user] : $row->rev_user_text;
+		}
+
+		return $users;
```

Comment:
should we rewind `$res` here post iteration?

Reply:
I'm not sure, I spotted it somewhere, that's why I used that
:bell: shame

Reply:
let's rewind it before return, just in case